app/navigation/navigation-items.service.ts

```diff
@@ -11,8 +11,8 @@ export class NavigationItemsService {
         return this.menuItems;
     }
 
-    public addMenuItem(menuItem : MenuItem) {
-        this.menuItems.push(menuItem);
+    public addMenuItem(label: string, icon: string, routerLinkParameters: any[]) {
```

Comment:
Could we keep the `MenuItem` type as parameter? 
Put an additional line in "navigation.module.ts": `export * from "./menu-item";` so it gets passed on.

Code using the service then needs to `import {MenuItem} from "ndb-core"` but I feel this would be a little cleaner?